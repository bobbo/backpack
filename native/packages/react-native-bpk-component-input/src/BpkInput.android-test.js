/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2017 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import BpkInput from './BpkInput';

describe('Android', () => {
  describe('BpkInput', () => {
    it('should render correctly', () => {
      const tree = renderer.create(
        <BpkInput text={'Input_text'} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render placeholder correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={''}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render small correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          small
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render small placeholder correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={''}
          small
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render valid correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          valid
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render small valid correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          valid
          small
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render invalid correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          valid={false}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render small invalid correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          valid={false}
          small
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render disabled correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          disabled
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render small disabled correctly', () => {
      const tree = renderer.create(
        <BpkInput
          placeholder={'placeholder_text'}
          text={'Input_text'}
          disabled
          small
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should apply user-styling correctly', () => {
      const tree = renderer.create(
        <BpkInput
          style={{ width: 200, backgroundColor: 'blue' }}
          placeholder={'placeholder_text'}
          text={'Input_text'}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
