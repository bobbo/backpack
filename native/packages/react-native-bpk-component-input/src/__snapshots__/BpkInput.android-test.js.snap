// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Android BpkInput should apply user-styling correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "backgroundColor": "blue",
        "width": 200,
      },
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
</View>
`;

exports[`Android BpkInput should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
</View>
`;

exports[`Android BpkInput should render disabled correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
    ]
  }
>
  <TextInput
    editable={false}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "color": "rgb(230, 228, 235)",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
</View>
`;

exports[`Android BpkInput should render invalid correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
  />
</View>
`;

exports[`Android BpkInput should render placeholder correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontStyle": "italic",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`Android BpkInput should render small correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "height": 32,
        "padding": 8,
      },
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "400",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
</View>
`;

exports[`Android BpkInput should render small disabled correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "height": 32,
        "padding": 8,
      },
    ]
  }
>
  <TextInput
    editable={false}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "400",
        },
        Object {
          "color": "rgb(230, 228, 235)",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
</View>
`;

exports[`Android BpkInput should render small invalid correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "height": 32,
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "400",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
        Object {
          "height": 16,
          "width": 16,
        },
      ]
    }
  />
</View>
`;

exports[`Android BpkInput should render small placeholder correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "height": 32,
        "padding": 8,
      },
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "400",
        },
        Object {
          "fontStyle": "italic",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`Android BpkInput should render small valid correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      Object {
        "height": 32,
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
        Object {
          "fontFamily": "System",
          "fontSize": 11,
          "fontWeight": "400",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
        Object {
          "height": 16,
          "width": 16,
        },
      ]
    }
  />
</View>
`;

exports[`Android BpkInput should render valid correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgb(230, 228, 235)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "padding": 12,
        "width": "auto",
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    padding={0}
    placeholder="placeholder_text"
    placeholderTextColor="rgb(178, 174, 189)"
    style={
      Array [
        Object {
          "color": "rgb(82, 76, 97)",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "width": "auto",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="Input_text"
  />
  <Image
    source={1}
    style={
      Array [
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
  />
</View>
`;
